### ISSUE TITLE
Server error when applying a dpdk patch with git-pw

### ISSUE THREAD

**Issue by david-marchand** [getpatchwork/patchwork#260]
Trying to apply a dpdk patch triggers a 500 error.

```
[pw]
        server = https://patchwork.dpdk.org/api/1.0
        token = XXX
        project = dpdk
```

```
$ git remote -v |grep origin
origin	git://dpdk.org/dpdk (fetch)
origin	git://dpdk.org/dpdk (push)
```

```
$ git-pw --version
git-pw, version 1.5.0
```

```
$ git-pw patch apply 51394
Starting new HTTPS connection (1): patchwork.dpdk.org
Server error. Please report this issue to https://github.com/getpatchwork/patchwork
Traceback (most recent call last):
  File "/usr/bin/git-pw", line 11, in <module>
    sys.exit(cli())
  File "/usr/lib64/python2.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib64/python2.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/lib64/python2.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib64/python2.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib64/python2.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib64/python2.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/git_pw/patch.py", line 40, in apply_cmd
    patch = api.detail('patches', patch_id)
  File "/usr/lib/python2.7/site-packages/git_pw/api.py", line 253, in detail
    return get(url, params, stream=False).json()
  File "/usr/lib/python2.7/site-packages/git_pw/api.py", line 149, in get
    _handle_error('fetch', exc)
  File "/usr/lib/python2.7/site-packages/git_pw/api.py", line 147, in get
    rsp.raise_for_status()
  File "/usr/lib/python2.7/site-packages/requests/models.py", line 834, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error
```

**Comment by david-marchand** [comment]
I dag a little more, this patch has been resent by the author and has format issues.

http://patchwork.dpdk.org/patch/50869/
http://patchwork.dpdk.org/patch/51394/

**Comment by stephenfin** [comment]
@tmonjalo Any chance we could get logs from Patchwork when attempting to download this patch?

**Comment by tmonjalo** [comment]
Internal Server Error: /api/1.0/patches/51394/
Traceback (most recent call last):
  File "/usr/local/virtualenv-patchwork-2.0/local/lib/python2.7/site-packages/django/core/handlers/exception.py", line 39, in inner
    response = get_response(request)
  File "/usr/local/virtualenv-patchwork-2.0/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/virtualenv-patchwork-2.0/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/virtualenv-patchwork-2.0/local/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/virtualenv-patchwork-2.0/local/lib/python2.7/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/virtualenv-patchwork-2.0/local/lib/python2.7/site-packages/rest_framework/views.py", line 489, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/virtualenv-patchwork-2.0/local/lib/python2.7/site-packages/rest_framework/views.py", line 449, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/virtualenv-patchwork-2.0/local/lib/python2.7/site-packages/rest_framework/views.py", line 486, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/local/virtualenv-patchwork-2.0/local/lib/python2.7/site-packages/rest_framework/generics.py", line 254, in get
    return self.retrieve(request, *args, **kwargs)
  File "/usr/local/virtualenv-patchwork-2.0/local/lib/python2.7/site-packages/rest_framework/mixins.py", line 58, in retrieve
    return Response(serializer.data)
  File "/usr/local/virtualenv-patchwork-2.0/local/lib/python2.7/site-packages/rest_framework/serializers.py", line 534, in data
    ret = super(Serializer, self).data
  File "/usr/local/virtualenv-patchwork-2.0/local/lib/python2.7/site-packages/rest_framework/serializers.py", line 263, in data
    self._data = self.to_representation(self.instance)
  File "/usr/local/virtualenv-patchwork-2.0/local/lib/python2.7/site-packages/rest_framework/serializers.py", line 501, in to_representation
    ret[field.field_name] = field.to_representation(attribute)
  File "/usr/local/virtualenv-patchwork-2.0/local/lib/python2.7/site-packages/rest_framework/fields.py", line 1755, in to_representation
    return method(value)
  File "/srv/patchwork/patchwork/api/patch.py", line 131, in get_prefixes
    return clean_subject(instance.name)[1]
  File "/srv/patchwork/patchwork/parser.py", line 645, in clean_subject
    subject = clean_header(subject)
  File "/srv/patchwork/patchwork/parser.py", line 149, in clean_header
    sane_header = sanitise_header(header)
  File "/srv/patchwork/patchwork/parser.py", line 75, in sanitise_header
    value = decode_header(header_contents)
  File "/usr/lib/python2.7/email/header.py", line 73, in decode_header
    header = str(header)
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2018' in position 55: ordinal not in range(128)

**Comment by stephenfin** [comment]
@tmonjalo++ Looks like this was resolved in a128310bf and related patches by @daxtens. We're not really supporting the `stable/2.0` branch any more so you might want to backport those patches yourself (or upgrade to v2.1)

**Comment by stephenfin** [comment]
Closing as this is resolved in recent releases
